core/src/tile/tileManager.cpp

```diff
@@ -14,16 +14,18 @@
 
 namespace Tangram {
 
-TileManager::TileManager(TileTaskQueue& _tileWorker) : m_workers(_tileWorker) {
+TileManager::TileManager(std::shared_ptr<TileTaskQueue> _tileWorker) : m_workers(_tileWorker) {
 
     m_tileCache = std::unique_ptr<TileCache>(new TileCache(DEFAULT_CACHE_SIZE));
 
     m_dataCallback = TileTaskCb{[this](std::shared_ptr<TileTask>&& task) {
+        if (m_workers) {
             if (task->hasData()) {
-                m_workers.enqueue(std::move(task));
+                m_workers->enqueue(std::move(task));
             } else {
                 task->cancel();
             }
```

Comment:
Even a weak pointer won't help I guess. When the weak<TileWorker> is upgraded to shared<TileWorker> in TileTaskCb it can be the last handle to TileWorker which then may get destructed on a download thread. Since the TileWorker has a shared_ptr to Scene this is not good.

How about testing ```if (task->isCanceled()) { return; }``` in  m_dataCallback first?

All tasks will be canceled in the TileManager destructor